File: groupAnagrams/groupAnagrams.cpp


#include "../include/include.hpp"


class Solution {
public:
	vector<vector<string>> groupAnagrams(vector<string>& strs)
	{
		vector<vector<string>>	res;
		unordered_map<string, vector<string>> anagMap;

		for (auto str : strs)
		{
			string sorted = str;
			sort(sorted.begin(), sorted.end());
			anagMap[sorted].push_back(str); 
		}

		for (auto &pair : anagMap)
		{
			res.push_back(pair.second);
		}

		return (res);
	}

};


int main()
{
	Solution ss;
	vector<string> strs = {"eat","tea","tan","ate","nat","bat"};


	print_nested_container(ss.groupAnagrams(strs));
	return (0);
}

File: groupAnagrams/groupAnagrams.py
class Solution:
    def groupAnagrams(self, strs):
        Map = {}

        for s in strs:
            sorted_s = "".join(sorted(s))
            Map[sorted_s] = Map.get(sorted_s,[]) + [s]

        res = []
        for key, value in Map.items():
            res.append(value)

        return res




ss = Solution()

print(ss.groupAnagrams(["eat","tea","tan","ate","nat","bat"]))